File: javascript/high-scores/high-scores.js
export class HighScores {
  constructor(arr) {
    this.arr = arr
  }

  get scores() {
    return this.arr
  }

  get latest() {
    return this.arr.reverse()[0]
  }

  get personalBest() {
    return Math.max(...this.arr)
  }

  get personalTopThree() {
    let empty = []
    let newArr = this.arr.sort((a,b) => a - b).reverse()
    let len = (this.arr.length < 3) ? this.arr.length : 3
    for ( let i = 0; i < len; i++){
      empty.push(newArr[i])
    }
    return empty
  }
}

File: python/high-scores/high_scores.py
class HighScores(object):
    def __init__(self, scores):
        self.scores = scores

    def latest(self):
        return self.scores[-1]

    def personal_best(self):
        return max(self.scores)

    def personal_top(self):
        return sorted(self.scores, reverse=True)[0:3]

    def report(self):
        if self.latest() == self.personal_best():
            return "Your latest score was {:d}. That's your personal best!".format(self.latest())

        else:
            diff = self.personal_best() - self.latest()
            return "Your latest score was {:d}. That's {:d} short of your personal best!".format(self.latest(), diff)
